bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/rich/pot/sp/RichProgram.scala

```diff
@@ -1,34 +1,30 @@
 package edu.gemini.spModel.rich.pot.sp
 
+import edu.gemini.pot.sp._
 import edu.gemini.spModel.gemini.obscomp.SPProgram
 import scala.collection.JavaConverters._
-import edu.gemini.pot.sp.{ISPObservation, ISPProgram}
 
 class RichProgram(prog:ISPProgram) {
   def spProgram:Option[SPProgram] = prog.dataObject.map(_.asInstanceOf[SPProgram])
 
-  def update(f:SPProgram => Unit) {
-    spProgram foreach {dataObj =>
+  def update(f:SPProgram => Unit): Unit =
+    spProgram.foreach { dataObj =>
       f(dataObj)
       prog.dataObject = dataObj
     }
-  }
 
   def allObservations:List[ISPObservation] = prog.getAllObservations.asScala.toList
 
   def obsByLibraryId(lid:String):Either[String, ISPObservation] =
-    allObservations.find(_.libraryId.map(_ == lid).getOrElse(false)).toRight("Observation with library id '%s' was not found.".format(lid))
+    allObservations.find(_.libraryId.exists(_ == lid)).toRight("Observation with library id '%s' was not found.".format(lid))
```

Comment:
You're right, I should have updated that along the way.  I touched this file to add a method only to later repent and take it back out.  I did a few obvious improvements along the way but I probably should have just rolled back all the changes to keep it from distracting from the rest.